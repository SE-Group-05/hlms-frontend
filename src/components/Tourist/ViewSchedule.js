import React, { useState } from 'react';
import {
  Input,
  Text,
  Stack,
  FormControl,
  Button,
  Heading,
  Box,
  Alert,
  AlertIcon,
  Checkbox,
} from '@chakra-ui/react';

import axios from 'axios';

const BASE_URL = '';
function ViewSchedule() {
  const [data, setData] = useState({
    methods: [],
    date: '',
    time: '',
    mark_completed: '',
  });
  //   const methods = [
  //     { value: 'Bus', label: 'Bus' },
  //     { value: 'Van', label: 'Van' },
  //     { value: 'Car', label: 'Car' },
  //   ];
  //const[data, setData]=useState([]);

  const [hasError, setHasError] = useState(false);
  const [editMode, setEditMode] = useState(false);
  const [show, setShow] = React.useState(false);


  const handleSubmit = async event => {
    event.preventDefault();
    console.log(data);
    try {
      var res = await axios.post(`${BASE_URL}/places`, JSON.stringify(data));
      console.log(res.status);
      console.log(res.data);
    } catch (e) {
      console.error(e);
    } finally {
      console.log('We do cleanup here');
    }
  };
  const handleChange = event => {
    var value = event.target.value;
    setData({ ...data, [event.target.name]: value });
  };

  return (
    <Box
      border="1px"
      borderColor="gray.200"
      p={['5px', '15px', '25px']}
      width="500px"
    >
      <Heading as="h4" size="md" textAlign="center" pb="20px" pt="5px">
        Current Schedule
      </Heading>
      {hasError ? (
        <Alert status="error" mb="20px">
          <AlertIcon />
          Error
        </Alert>
      ) : null}
      <form action="submit">
        <Stack spacing={3}>
          {editMode ? (
            <>
              <Text>Travelling destination</Text>
              <FormControl isRequired>
                <Input
                data-testid = "place"
                  type="text"
                  value={data.place}
                  name="place"
                  placeholder="Visiting Place"
                  aria-label="Place"
                  onChange={event => handleChange(event)}
                />
              </FormControl>
            </>
          ) : (
            <></>
          )}{' '}
          {editMode ? (
            <>
              <Text>Date</Text>
              <FormControl isRequired>
                <Input
                  data-testid = "date"
                  value={data.date}
                  name="date"
                  placeholder="Date"
                  aria-label="Date"
                  onChange={event => handleChange(event)}
                />
              </FormControl>
            </>
          ) : (
            <></>
          )}
          {editMode ? (
            <>
              <Text>Time</Text>
              <FormControl isRequired>
                <Input
                data-testid = "time"
                  type="time"
                  value={data.time}
                  name="time"
                  placeholder="Time"
                  aria-label="Time"
                  onChange={event => handleChange(event)}
                />
              </FormControl>
            </>
          ) : (
            <></>
          )}
          {editMode ? (
            <>
              <Text>Travelling Method</Text>

              <select>
                {/* value={useState.methods.value}
                onChange={this.handleChange} */}
              </select>
            </>
          ) : (
            <></>
          )}
          {editMode ? (
            <>
              <Text>Mark as completed</Text>
              <FormControl isRequired>
                <Checkbox
                  value={data.mark_completed}
                  name="mark_completed"
                  onChange={event => handleChange(event)}
                  errorBorderColor="crimson"
                />
              </FormControl>
            </>
          ) : (
            <></>
          )}
          {editMode ? null : (
            <Button colorScheme="teal" onClick={() => setEditMode(true)}>
              Edit
            </Button>
          )}
          {editMode ? (
            <Button
              colorScheme="cyan"
              type="submit"
              onClick={event => handleSubmit(event)}
            >
              Update the schedule
            </Button>
          ) : null}
          <Button
            colorScheme="red"
            type="submit"
            onClick={event => handleSubmit(event)}
          >
            Delete
          </Button>
        </Stack>
      </form>
    </Box>
  );
}

export default ViewSchedule;
