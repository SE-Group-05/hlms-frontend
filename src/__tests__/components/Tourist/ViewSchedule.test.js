// import dependencies
import React from "react";

// import react-testing methods
import { render, fireEvent, cleanup, waitFor } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import ViewSchedule from "../../../components/Tourist/ViewSchedule";

afterEach(cleanup);


it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <ViewSchedule />,
    div
  );
});


it("edit button works correctly", async () => {
  const { queryByTestId, getByText } = render(
    <ViewSchedule />
  );


  
  fireEvent.click(getByText("Edit")) 



  fireEvent.change(queryByTestId("place"),{ target: { value: 'mirissa' } })
  fireEvent.change(queryByTestId("date"),{ target: { value: '2020/12/23' } })
  fireEvent.change(queryByTestId("time"),{ target: { value: '23:20' } })

 
  
  fireEvent.click(getByText("Delete")) 

  
});

it("matches snapshot", () => {
  const tree = renderer.create(
    <ViewSchedule
      data={{
        placename: "Galle",
        distance: "20",
        timeToReach: "30",
        description: "no description",
        methods: ["bus"],
      }}
    />
  );

  expect(tree).toMatchSnapshot();
});
