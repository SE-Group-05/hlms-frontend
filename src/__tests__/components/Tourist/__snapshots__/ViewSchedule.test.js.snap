// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-1ceyi20"
>
  <h4
    className="chakra-heading css-biiv2x"
  >
    Current Schedule
  </h4>
  <form
    action="submit"
  >
    <div
      className="chakra-stack css-1n2jfo2"
    >
      <button
        className="chakra-button css-oqfhsh"
        onClick={[Function]}
        type="button"
      >
        Edit
      </button>
      <button
        className="chakra-button css-oqfhsh"
        onClick={[Function]}
        type="submit"
      >
        Delete
      </button>
    </div>
  </form>
</div>
`;
